supabase/_async/client.py

```diff
@@ -0,0 +1,289 @@
+
+    @deprecated("1.1.1", "1.3.0", details="Use `.functions` instead")
+    def functions(self) -> AsyncFunctionsClient:
+        return AsyncFunctionsClient(self.functions_url, self._get_auth_headers())
+
+    def table(self, table_name: str) -> AsyncRequestBuilder:
+        """Perform a table operation.
+
+        Note that the supabase client uses the `from` method, but in Python,
+        this is a reserved keyword, so we have elected to use the name `table`.
+        Alternatively you can use the `.from_()` method.
+        """
+        return self.from_(table_name)
+
+    def from_(self, table_name: str) -> AsyncRequestBuilder:
+        """Perform a table operation.
+
+        See the `table` method.
+        """
+        return self.postgrest.from_(table_name)
+
+    def rpc(self, fn: str, params: Dict[Any, Any]) -> AsyncFilterRequestBuilder:
+        """Performs a stored procedure call.
+
+        Parameters
+        ----------
+        fn : callable
+            The stored procedure call to be executed.
+        params : dict of any
+            Parameters passed into the stored procedure call.
+
+        Returns
+        -------
+        SyncFilterRequestBuilder
+            Returns a filter builder. This lets you apply filters on the response
+            of an RPC.
+        """
+        return self.postgrest.rpc(fn, params)
+
+    @property
+    def postgrest(self):
+        if self._postgrest is None:
+            self.options.headers.update(self._get_token_header())
+            self._postgrest = self._init_postgrest_client(
+                rest_url=self.rest_url,
+                headers=self.options.headers,
+                schema=self.options.schema,
+                timeout=self.options.postgrest_client_timeout,
+            )
+        return self._postgrest
+
+    @property
+    def storage(self):
+        if self._storage is None:
+            headers = self._get_auth_headers()
+            headers.update(self._get_token_header())
+            self._storage = self._init_storage_client(
+                storage_url=self.storage_url,
+                headers=headers,
+                storage_client_timeout=self.options.storage_client_timeout,
+            )
+        return self._storage
+
+    @property
+    def functions(self):
+        if self._functions is None:
+            headers = self._get_auth_headers()
+            headers.update(self._get_token_header())
+            self._functions = AsyncFunctionsClient(self.functions_url, headers)
+        return self._functions
+
+    #     async def remove_subscription_helper(resolve):
+    #         try:
+    #             await self._close_subscription(subscription)
+    #             open_subscriptions = len(self.get_subscriptions())
+    #             if not open_subscriptions:
+    #                 error = await self.realtime.disconnect()
+    #                 if error:
+    #                     return {"error": None, "data": { open_subscriptions}}
```

Comment:
We can probably delete this tbh not sure when we'll be reinstating it if at all

Reply:
I'll leave this for now as I'm working to get realtime back into the library now.